### PR TITLE
feat(redux): added Redux Functionality to project

### LINKED ISSUE #93: Redux Integration [blocked on API Integration]
[pick this task after API integration using context is done]

Each can pick one module and migrate from context to redux

Introduce redux library learn more structure way to manage data

### PR DESCRIPTION
**Pull request by smilegupta**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -12,8 +12,24 @@ import { NotificationProvider } from "notification";
...
+    // State Variables
+    const dispatch = useDispatch();
+
+    // useEffect Hook
+    useEffect(() => {
+        fetchData();
+    });
+
+    // Function to Fetch Data
+    const fetchData = useCallback(async () => {
```
Comment (rparsh) [inline src/App.js:28]: why using async function ?

File: src/pages/Customers/AddCustomer.js [hunk 2]
```diff
@@ -37,8 +37,11 @@ const AddCustomers = () => {
                     fields
                 );
                 if (error || response.error) {
+                    console.log(response.error.description);
```
Comment (rparsh) [inline src/pages/Customers/AddCustomer.js:40]: remove console

File: src/redux/actions/index.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { ActionTypes } from "../constants/actions-types";
```
Comment (rparsh) [inline src/redux/actions/index.js:1]: use absolute import

File: src/redux/reducers/index.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { itemsList } from "./itemReducer";
```
Comment (rparsh) [inline src/redux/reducers/index.js:1]: use absolute import everywhere

File: src/App.js [hunk 5]
```diff
@@ -12,8 +12,24 @@ import { NotificationProvider } from "notification";
...
 import ErrorBoundary from "common/ErrorBoundary";
+import { useDispatch } from "react-redux";
+import { fetchCustomersList, fetchItemsList } from "redux/actions/index";
 
 function App() {
+    // State Variables
+    const dispatch = useDispatch();
+
+    // useEffect Hook
+    useEffect(() => {
```
Comment (rparsh) [inline src/App.js:23]: this will trigger on every state/prop update

File: src/App.js (hunk 1 above)
Comment (smilegupta) [reply src/App.js]: Removed it.

**Review by InvincibleZeal — APPROVED** [review]

**Pull request merged**